src/compat/posix/proc/exec.c

```diff
@@ -55,18 +57,10 @@ int exec_call(void) {
 		const struct cmd *cmd;
 
 		cmd = cmd_lookup(cmd_name);
-
```

Comment:
Why you remove this?
Revert please

Reply:
I removed it because cmd_exec checks if the cmd exists anyways, so found that this check is redundant. Should I revert back though?

Reply:
Yes, I see. But I don't understand why did you move checking from here? Why did you place it in cmd2mod(). why did you move it at all?

Reply:
My aim was to remove setting errno to ENOENT in the exec_call() function, so I removed the check in exec_call() and simply used the check already present in cmd_exec(). 
On running however, as cmd was NULL, cmd2mod() caused errors as it was trying to perform its operation on a NULL pointer. So I included the check there too.
So should I put the check back here and remove the check in cmd2mod()?

Reply:
> My aim was to remove setting ecode and errno to ENOENT
Why? :)

I thought your aim is "execv returns 0 and doesn't set errno" :)

Reply:
> But Why do you want to differ in two situations
>
> When cmd_exec return with errors
> When cmd has not been found
> I supposed You can just return ecode from exec_call and then return it in execv

I thought when you said this in the earlier comment, you wanted me to set all errors in the cmd_exec() function itself, my bad. Should I revert it back to the way it was?

Reply:
> Should I revert it back to the way it was?
Yes please.
I'm sorry that uncleanly put it

Reply:
Done :)